src/ctest.ts

```diff
@@ -188,9 +190,62 @@ export function parseCatchTestOutput(output: string): FailingTestDecoration[] {
   return decorations;
 }
 
+/**
+ * [Unity framework](https://github.com/ThrowTheSwitch/Unity) output parsing
+ *
+ * @param output   Unity output
+ *
+ * @returns        Decorations for failing tests that were discovered during parsing
+ *
+ * Sample output:
+ * "..!.../../test.c:23:FAIL: Expected 1 was 0"
+ * * `..!.`: each dot marks a test that was previously ran, each exclamation mark marks an ignored test
+ * * `../../test.c`: test suite file name. I expect it to always be relative to build directory so it starts with
+ *                   `../`. Otherwise it will be very hard to parse the output because of the dots.
+ *                   It's possible to disable the number of ran tests indication using `-s` flag but
+ *                   that's again another limitation. In most of the cases tests will be ran without
+ *                   additional arguments and build directory won't be mixed with sources.
+ * * `23`: line number.
+ * * `FAIL`: failure indication. It could also be `PASS` or `IGNORE`.
+ * * `Expected 1 was 0`: assertion message. Should always be on one line expect the cases when user
+ *                       explicitly puts the `\n` in the message. That shouldn't typically happen
+ *                       and looks like a known limitation to me.
+ */
+export function parseUnityFixtureOutput(output: string): FailingTestDecoration[] {
+  const lines_with_ws = output.split('\n');
+  const lines = lines_with_ws.map(l => l.trim());
+  const decorations: FailingTestDecoration[] = [];
+  const failedTestRegex = /^([\.!]*)(\.\..*):(\d+):(.*):FAIL: (.*)/;
+
+  for (const line of lines) {
+    const parseRes = failedTestRegex.exec(line);
+    if (parseRes) {
+      const [_all, testsRanStr, file, lineNumberStr, testName, message] = parseRes;
```

Comment:
I'm pretty sure you can do:
```
const [, , file, lineNumberStr, , message] = parseRes;
```
or better:
```
const [/*all*/, /*testsRanStr*/, file, lineNumberStr, /*testName*/, message] = parseRes;
```
instead.  (I do see that the code you're modelling this on doesn't do this, but ownership of the code has changed since that was written 😄)

Reply:
Thanks, that'll simplify the code! I prefer the second option -- that'll make it easy to spot those parameters if they're ever required.